sdk/go/common/resource/plugin/host.go

```diff
@@ -89,13 +89,17 @@ type Host interface {
 	// called before (e.g.) hard-closing any gRPC connection.
 	SignalCancellation() error
 
+	// StartDebugging asks the host to start a debugging session with the given configuration.
+	StartDebugging(DebuggingInfo) error
```

Comment:
Adding a new method to an interface like this is a breaking change, so upgrades are going to take longer than normal as we now need to manually implement this in a lot of places.

In the future I suggest implementing this as a standalone interface, and then test the host to see whether it supports debugging when you need it.

```go
type Host interface { /* unchanged */ }

type Debugger interface {
    StartDebugging(DebuggingInfo) error
}

...

func (eng *hostServer) StartDebugger(ctx context.Context,
 	req *pulumirpc.StartDebuggerRequest,
 ) (*emptypb.Empty, error) {
    debugger, ok := eng.host.(Debugger)
    if !ok {
        return &emptypb.Empty{}, nil
    }

 	err := debugger.StartDebugging(info)
 	if err != nil {
 		return nil, err
 	}
    return &emptypb.Empty{}, nil
}
```